lib/config/global.ts

```diff
@@ -1,40 +1,56 @@
 import type { RenovateConfig, RepoGlobalConfig } from './types';
 
-let repoGlobalConfig: RepoGlobalConfig = {};
-
-// TODO: once global config work is complete, add a test to make sure this list includes all options with globalOnly=true (#9603)
-const repoGlobalOptions = [
-  'allowCustomCrateRegistries',
-  'allowPlugins',
-  'allowPostUpgradeCommandTemplating',
-  'allowScripts',
-  'allowedPostUpgradeCommands',
-  'binarySource',
-  'customEnvVariables',
-  'dockerChildPrefix',
-  'dockerImagePrefix',
-  'dockerUser',
-  'dryRun',
-  'exposeAllEnv',
-  'migratePresets',
-  'privateKey',
-  'privateKeyOld',
-  'localDir',
-  'cacheDir',
-];
-
-export function setGlobalConfig(
-  config: RenovateConfig | RepoGlobalConfig = {}
-): RenovateConfig {
-  repoGlobalConfig = {};
-  const result = { ...config };
-  for (const option of repoGlobalOptions) {
-    repoGlobalConfig[option] = config[option];
-    delete result[option];
+export class GlobalConfig {
+  // TODO: once global config work is complete, add a test to make sure this list includes all options with globalOnly=true (#9603)
+  private static readonly OPTIONS = [
+    'allowCustomCrateRegistries',
+    'allowedPostUpgradeCommands',
+    'allowPlugins',
+    'allowPostUpgradeCommandTemplating',
+    'allowScripts',
+    'binarySource',
+    'cacheDir',
+    'customEnvVariables',
+    'dockerChildPrefix',
+    'dockerImagePrefix',
+    'dockerUser',
+    'dryRun',
+    'exposeAllEnv',
+    'localDir',
+    'migratePresets',
+    'privateKey',
+    'privateKeyOld',
+  ];
+
+  private static config: RepoGlobalConfig = {};
+
+  static get(): RepoGlobalConfig;
+  static get<Key extends keyof RepoGlobalConfig>(
+    key?: Key
+  ): RepoGlobalConfig[Key];
+  static get<Key extends keyof RepoGlobalConfig>(
+    key?: Key
+  ): RepoGlobalConfig | RepoGlobalConfig[Key] {
+    return key ? GlobalConfig.config[key] : GlobalConfig.config;
   }
-  return result;
-}
 
-export function getGlobalConfig(): RepoGlobalConfig {
-  return repoGlobalConfig;
+  static set(config: RenovateConfig | RepoGlobalConfig): RenovateConfig {
+    GlobalConfig.reset();
+
+    const result = { ...config };
+    for (const option of GlobalConfig.OPTIONS) {
+      GlobalConfig.config[option] = config[option];
+      delete result[option];
+    }
+
+    return result;
+  }
+
+  static reset(): void {
+    GlobalConfig.config = {};
+  }
+
+  static get isDryRun(): boolean {
+    return GlobalConfig.get('dryRun');
+  }
```

Comment:
Do we really need this?

Reply:
Not really, It is optional thing

Reply:
i don't think we need this, as `GlobalConfig.get('dryRun')` isn't much longer than `GlobalConfig.isDryRun`. So i prefer to remove this